src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/controller/CertificateApi.kt

```diff
@@ -52,6 +51,25 @@ interface CertificateApi {
     )
     @PostMapping("/certificate/document")
     @PostExchange("/certificate/document")
-    fun setCertificateDocument(@RequestBody certificateDocumentRequestDto: CertificateDocumentRequestDto):ResponseEntity<CertificateDocumentResponseDto>
+    fun setCertificateDocument(@RequestBody certificateDocumentRequestDto: CertificateDocumentRequestDto): ResponseEntity<CertificateDocumentResponseDto>
+
+
+    @Operation(
+        summary = "Request a specific certificate document for a given BPN.",
+        operationId = "retrieveCertificateDocument",
+        description = "This endpoint call to request a specific certificate document for a given BPN.",
+        responses = [
+            ApiResponse(responseCode = "200", description = "Document for the given ID"),
+            ApiResponse(responseCode = "400", description = "Malformed URL", content = [Content()]),
+            ApiResponse(responseCode = "401", description = "Unauthorized", content = [Content()]),
+            ApiResponse(responseCode = "404", description = "cdID not found", content = [Content()]),
```

Comment:
Can we please make error description something like 'Certificate Document ID not found.' and handle the exception using CertificateControllerExceptionHandler